Dockerfiles/agent/entrypoint.d/otel-agent

```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+if [ -f /opt/datadog-agent/embedded/bin/otel-agent ] && [ -x /opt/datadog-agent/embedded/bin/otel-agent ]; then
 exec /opt/datadog-agent/embedded/bin/otel-agent --config /etc/datadog-agent/otel-config.yaml
+fi
```

Comment:
This script will be invoked if and only if the agent docker container is explicitly invoked (by [`entrypoint.sh`](https://github.com/DataDog/datadog-agent/blob/main/Dockerfiles/agent/entrypoint.sh)) with `ENTRYPOINT=otel-agent`.
If a user explicitly specify `ENTRYPOINT=otel-agent`, it’s most probably on purpose and silently exiting without any log or error might be surprising.

If `ENTRYPOINT=otel-agent` is set on a setup (like in a Kubernetes pod) where exiting containers are systematically restarted, because this script will exit immediately if `otel-agent` cannot be found, the container will be endlessly restarted and end up in `CrashLoopBackoff` state.

If the goal is to cope with the fact that some agent docker images have an `otel-agent` and some other haven’t, maybe the best is to not have this entrypoint script at all in the docker images that don’t have `otel-agent`.

Reply:
Is it expected to have the entrypoint set as `otel-agent`?

Reply:
Checked together, all good 👍